src/index.js

```diff
@@ -60,14 +60,86 @@ const getVersion = memoize(async () => (await getPackage()).version || 'n/a');
 
 const getName = memoize(async () => (await getPackage()).name || 'n/a');
 
+const getFetchContext = memoize(() => {
+  /* istanbul ignore next */
+  if (process.env.HELIX_FETCH_FORCE_HTTP1) {
+    return fetchAPI.context({
+      httpProtocol: 'http1',
+      httpsProtocols: ['http1'],
+    });
+  } else {
+    /* istanbul ignore next */
+    return fetchAPI;
+  }
+});
+
+/**
+ * Use fetch to emulate a request call as it is used in this code.
+ * @param {object} opts Request options.
+ * @returns {object} a response object.
+ */
+async function request(opts) {
+  const { uri } = opts;
+  if (!uri) {
+    throw new Error('request needs uri parameter.');
+  }
+  const options = {
+    cache: 'no-store',
+    ...opts,
+  };
+
+  delete options.uri;
+  const context = getFetchContext();
+
+  if (options.timeout) {
+    options.signal = context.timeoutSignal(options.timeout);
+    delete options.timeout;
+  }
+
+  const start = Date.now();
+  let response;
+  let body;
+  try {
+    response = await context.fetch(uri, options);
+    body = await response.text();
+  } catch (e) {
+    /* istanbul ignore next */
+    if (e instanceof fetchAPI.AbortError) {
+      e.options = options;
```

Comment:
this is a bit ugly, but I didn't want to change the error handling further down...